### PR TITLE
Added confirmation question validation

### PR DESCRIPTION
**Pull request by pricea1**
### What is the context of this PR?
Confirmation question needs to validate on question title and option labels.

### How to review 
1. Create confirmation page
2. Ensure validation error displays if question title missing
3. Ensure validation error displays if radio option label missing
4. Ensure validation error displays if radio option labels are the same

Observe error count in navigation changing appropriately

### CODE REVIEW

**Comment by pricea1** [comment]
[Validator - Playback pages (2)](https://trello.com/c/Zfg1kCJg/1176-validator-playback-pages-2)

**Comment by SamGodwin2** [comment]
Screenshot 2019-08-06 at 09 08 04
Bit of a weird stating state, when you first create a confirmation question the error is that they both have the same value would it be possible to make the is required error have higher priority?

Also when changing one of the labels the error leaves the other even when the label is empty as seen in the snapshot.

**Comment by pricea1** [comment]
> Screenshot 2019-08-06 at 09 08 04
> Bit of a weird stating state, when you first create a confirmation question the error is that they both have the same value would it be possible to make the is required error have higher priority?
> 
> Also when changing one of the labels the error leaves the other even when the label is empty as seen in the snapshot.

Now fixed

File: eq-author/src/App/questionConfirmation/Design/ConfirmationOption.js [hunk 1]
```diff
@@ -57,6 +71,9 @@ ConfirmationOption.propTypes = {
   label: PropTypes.string.isRequired,
   onChange: PropTypes.func.isRequired,
   onUpdate: PropTypes.func.isRequired,
+  getValidationError: PropTypes.func.isRequired,
 };
 
-export default ConfirmationOption;
+export default flowRight(withValidationError(CONFIRMATION_OPTION))(
```
Comment (SamGodwin2) [inline eq-author/src/App/questionConfirmation/Design/ConfirmationOption.js]: Could we do `withValidationError(CONFIRMATION_OPTION)(UnwrappedConfirmationOption)` just to save importing flowRight?

File: eq-author/src/App/questionConfirmation/Design/Editor.js [hunk 2]
```diff
@@ -74,13 +82,19 @@ export class UnwrappedEditor extends React.Component {
...
       </Wrapper>
     );
   }
 }
 
-const withConfirmationEditing = flow(withEntityEditor("confirmation"));
+//const withConfirmationEditing = flow(withEntityEditor("confirmation"));
 
-export default withConfirmationEditing(UnwrappedEditor);
+//export default withConfirmationEditing(UnwrappedEditor);
```
Comment (SamGodwin2) [inline eq-author/src/App/questionConfirmation/Design/Editor.js]: Guessing these got left behind by accident. 😄

File: eq-author/src/graphql/fragments/question-confirmation.graphql [hunk 3]
```diff
@@ -3,12 +3,34 @@ fragment QuestionConfirmation on QuestionConfirmation {
...
+    validationErrorInfo {
+      id
+      errors {
+        id
+        type
+        field
+        errorCode
+      }
+      totalCount
+    }
```
Comment (SamGodwin2) [inline eq-author/src/graphql/fragments/question-confirmation.graphql]: Can we import the fragments here to compact this query a little further?

Comment (pricea1) [reply eq-author/src/graphql/fragments/question-confirmation.graphql]: Done.

File: eq-author-api/schema/resolvers/base.js [hunk 4]
```diff
@@ -433,8 +439,8 @@ const Resolvers = {
       const questionConfirmation = {
         id: uuid.v4(),
         title: "",
-        positive: { label: null, description: null },
-        negative: { label: null, description: null },
+        positive: { id: uuid.v4(), label: "", description: null },
+        negative: { id: uuid.v4(), label: "", description: null },
```
[the remark this answers is not in the document]
Comment (pricea1) [reply eq-author-api/schema/resolvers/base.js]: Now changed

File: eq-author/src/App/questionConfirmation/Design/ConfirmationOption.js (hunk 1 above)
Comment (pricea1) [reply eq-author/src/App/questionConfirmation/Design/ConfirmationOption.js]: Changed

File: eq-author/src/App/questionConfirmation/Design/Editor.js (hunk 2 above)
Comment (pricea1) [reply eq-author/src/App/questionConfirmation/Design/Editor.js]: yup. now removed

**Comment by pricea1** [comment]
> Clicking preview on a confirmation question breaks it.

Good spot - now fixed

File: eq-author/src/App/questionConfirmation/Design/Editor.js [hunk 5]
```diff
@@ -28,6 +30,7 @@ export class UnwrappedEditor extends React.Component {
     onChange: PropTypes.func.isRequired,
     onUpdate: PropTypes.func.isRequired,
     confirmation: propType(confirmationFragment).isRequired,
+    getValidationError: PropTypes.func,
```
Comment (heofs) [inline eq-author/src/App/questionConfirmation/Design/Editor.js]: Should this be `.isRequired`?

Comment (pricea1) [reply eq-author/src/App/questionConfirmation/Design/Editor.js]: I've changed it now.

**Pull request merged**